docs/adr/0003-outsource-user-management.md

```diff
@@ -0,0 +1,89 @@
+# 3. Outsource User Management
+
+* Status: accepted <!-- optional -->
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1 <!-- optional -->
+* Date: 2020-12-09 <!-- optional -->
+
+Technical Story: [Skip account-service by talking to CS3 user-api](https://github.com/owncloud/ocis/pull/1020) <!-- optional -->
+
+## Context and Problem Statement
+
+To attach metadata like shares to users ownCloud relies on persistent, non-reassignable, unique identifiers for users (and files). Email and username can change when a user changes his name. But even the OIDC sub+iss combination may change when the IdP changes. While there is [an account porting protocol](https://openid.net/specs/openid-connect-account-porting-1_0.html) that describes how a relying party such as ownCloud should should behave, it still requires the RP to maintain its own user identifiers.
+
+## Decision Drivers <!-- optional -->
+
+* OCIS should be a single binary that can run out of the box without external dependencies like an LDAP server.
+* Time: we want to build a release candidate asap.
+
+## Considered Options
+
+* Accounts service wraps LDAP
+* [GLauth](https://github.com/glauth/glauth) wraps accounts service
+
+## Decision Outcome
+
+Chosen option: "Move accounts functionality to GLAuth and name it accounts", by moving the existing accounts service file based persistence to GLAuth and use it as a drop in replacement for an LDAP server. The reverse index and web ui existing in the accounts service will move as well in order to make GLAuth a standalone, small scale user management with write capabilities.
+
+### Product summary
+- GLAuth is a drop in user management for small scale deployments that don't rely on an actual LDAP.
+- OCIS admins can either use the web ui to manage users in GLAuth or use existing tools in their IDM.
+- We hide the complexity by embedding an OpenID Provider, an LDAP server and a user management web ui.
+
+### Resulting deployment options
+- Single binary: admin can manage users, groups and roles using the built in web ui (GLAuth)
+- External LDAP: OCIS admin needs to use existing tool to manage users
```

Comment:
From my POV it would be great if we could really lay out the options that are available for operating with some bullet points (including requirements and consequences). Something like

a) Use internal user management
- Recommended for small scale use cases and simple deployments
- Users are stored and managed within ownCloud
- User roles are managed and assigned in ownCloud
- etc.

b) Use external user management
- Recommended for mid and large scale use cases
- Users are stored and managed within an external LDAP/AD directory
- User roles are managed in ownCloud
- User roles are assigned in an external IDM
- etc.

c) ?

Probably this is not all correct but I'm trying to give an idea.